dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/ProgramRuleEngine.java

```diff
@@ -131,14 +106,28 @@ public ProgramRuleEngine( ProgramRuleEntityMapperService programRuleEntityMapper
     {
         List<RuleEffect> ruleEffects = new ArrayList<>();
 
+        String programStageUid = programStageInstance != null ? programStageInstance.getProgramStage().getUid() : null;
+
+        List<ProgramRule> programRules = implementableRuleService
+            .getImplementableRules( program, programStageUid )
+            .stream()
+            .filter( rule -> Objects.isNull( rule.getProgramStage() ) ||
```

Comment:
Do we need to filter on program stage when we also supply it to the store/database query?

Reply:
Of course not :)